src/compiler/emitter.ts

```diff
@@ -4509,6 +4509,28 @@ namespace ts {
             return format & ListFormat.MultiLine ? 1 : 0;
         }
 
+        function siblingNodePositionsAreComparable(previousNode: Node, nextNode: Node) {
+            if (previousNode.kind === SyntaxKind.NotEmittedStatement && nextNode.kind === SyntaxKind.NotEmittedStatement) {
+                return false;
+            }
+            if (nodeIsSynthesized(previousNode) || nodeIsSynthesized(nextNode)) {
+                return false;
+            }
+
+            // Get the position next node and compare against nextNode. If they are not equal, nodes have been rearranged and positions cannot be compared.
+            const originalNextNode = getNodeAtPosition(currentSourceFile!, previousNode.end + 1);

```

Comment:
I believe this would be broken by spaces between the node end and the list separator, or by duplicate list separators:

```ts
import {
  , Foo
  , Bar
} from ""
```

`Foo`’s end is 16, but `Bar`’s pos is 20. 17–19 don’t appear in the span of either. This is why I speculated that you might have to look at tokens between nodes to see if they’re just list separators. Or maybe you can do something similar to what’s in `getNodeAtPosition` with `forEachChild` to handle this... that is, it should be easy to tell that the position you were given falls between two nodes, and you could choose to return the next one instead of the common parent. If you implement this correctly, I believe you should be able to drop the `+ 1`, which indicates the assumption that the list separator occurs immediately after `previousNode` and is exactly one character long.

Reply:
Also, whatever function this ends up using, it’s likely to be more expensive than the next couple conditions in this function, so this calculation should be moved to be the last.

Reply:
Makes sense. I have changed the algorithm for instead navigating the parent until the next specifier is found. I have limited to only look in the case the `previousNode` is an import specifier. Only for not affecting other scenarios.

Let me know what you think.

Reply:
I’m pretty confident in this logic being universal, so it would be nice to make it work for other scenarios and to keep such specific logic out of the emitter. To make sure I wasn’t giving you bad advice, I prototyped what I had in mind really quick and it seems to work, though the function isn’t beautiful. Here’s my attempt: https://github.com/andrewbranch/TypeScript/commit/c40b5790b4ed37373cba633fa42739f331ee6690

Reply:
One alternative more along the lines of what you have now would be to write a utility function that returns the index of a node in its parent’s list-like container. It would have to be a pretty big switch/case to determine what kind of node the parent is to access the right property. It would be more lines of code, but maybe a bit cleaner and more efficient than what I have here. A downside would be that it would need to be kept up to date when new node kinds are added in the future, but that doesn’t happen too often.